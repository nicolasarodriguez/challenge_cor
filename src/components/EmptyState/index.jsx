import React from 'react'
import { string } from 'prop-types'

const namespace = 'ui-emptystate'

const EmptyState = ({
  text,
  image
}) => {
  return (
    <div className={`${namespace}`}>
        <img
          className={`${namespace}__image`}
          src={process.env.PUBLIC_URL + '/assets/' + image}
          alt='empty state'
        />
        <p className={`${namespace}__text`}>{text}</p>
    </div>
  )
}

EmptyState.propTypes = {
  text: string.isRequired,
  image: string.isRequired
}

export default EmptyState
