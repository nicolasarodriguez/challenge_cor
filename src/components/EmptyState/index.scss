.emptystate {
    text-align: center;

    &__image {
        height: 250px;
        width: 250px;
    }

    &__text {
        font-size: 25px;
        font-weight: 600;
    }
}