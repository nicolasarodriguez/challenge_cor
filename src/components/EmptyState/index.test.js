import React from 'react';
import { render, screen } from '@testing-library/react';

import EmptyState from './index';

const defaultProps = {
  text: "No se encontro ningun resultado",
  image: "no-results.jpg"
}

const setup = (propOverrides = {}) => {
    const props = {
      ...defaultProps,
      ...propOverrides,
    };
  
    const view = render(<EmptyState {...props} />);
  
    return {
      props,
      view
    };
};

describe('<EmptyState />', () => {
  it('should match snapshot no results', () => {
      const { view } = setup();
      const test = screen.getByText('No se encontro ningun resultado');
      expect(test).toBeInTheDocument();
      expect(view).toMatchSnapshot();
  });

  it('should match snapshot no tasks', () => {
    const { view } = setup({
      text: 'No hay tareas cargadas',
      image: 'no-tasks.jpg'
    });
    const test = screen.getByText('No hay tareas cargadas');
    expect(test).toBeInTheDocument();
    expect(view).toMatchSnapshot();
});
});
