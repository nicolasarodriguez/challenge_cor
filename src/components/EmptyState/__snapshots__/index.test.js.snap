// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should match snapshot no results 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui-emptystate"
      >
        <img
          alt="empty state"
          class="ui-emptystate__image"
          src="/assets/no-results.jpg"
        />
        <p
          class="ui-emptystate__text"
        >
          No se encontro ningun resultado
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui-emptystate"
    >
      <img
        alt="empty state"
        class="ui-emptystate__image"
        src="/assets/no-results.jpg"
      />
      <p
        class="ui-emptystate__text"
      >
        No se encontro ningun resultado
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<EmptyState /> should match snapshot no tasks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui-emptystate"
      >
        <img
          alt="empty state"
          class="ui-emptystate__image"
          src="/assets/no-tasks.jpg"
        />
        <p
          class="ui-emptystate__text"
        >
          No hay tareas cargadas
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui-emptystate"
    >
      <img
        alt="empty state"
        class="ui-emptystate__image"
        src="/assets/no-tasks.jpg"
      />
      <p
        class="ui-emptystate__text"
      >
        No hay tareas cargadas
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
